src/v/s3/client.cc

```diff
@@ -101,8 +86,8 @@ static ss::sstring make_endpoint_url(
 }
 
 ss::future<configuration> configuration::make_configuration(
-  const public_key_str& pkey,
-  const private_key_str& skey,
+  const std::optional<public_key_str>& pkey,
+  const std::optional<private_key_str>& skey,
```

Comment:
you mentioned it in the commit message itself:

`cloud_storage,s3,v: optional keys, s3 pool uses cred applier, roles lib`

it looks like this commit needs to be split up into separate commits. generally we want to author PRs with commits such that each commit introduces a single logical change, so that those changes build on top of each other within the PR.

for example, it looks like in this PR there are

1. changes to make_configuration so that optional keys can be passed
2. changes to get_config that pass optional keys
3. introducing load / wait for credentials
4. using those credentials
5. changes to parse_timestamp
etc...